File: 3330. Find the Original Typed String I/possibleStringCount.js
/**
 * @param {string} word
 * @return {number}
 */
var possibleStringCount = function (word) {
    let chr = word[0]
    let res = 1

    for (let i = 1; i < word.length; i++) {
        if (word[i] === chr) {
            res++
        }
        else {
            chr = word[i]
        }
    }

    return res;
};

File: 3330. Find the Original Typed String I/possibleStringCount.cpp
#include <iostream>

using namespace std;

class Solution {
    public:
        int possibleStringCount(string word) {
            char chr = word[0];
            int res = 1;
    
            for(int i=1; i<word.size(); i++){
                if(word[i] == chr){
                    res++;
                }
                else{
                    chr = word[i];
                }
            }
            return res;
        }
    };